internalsdk/auth/auth.go

```diff
@@ -0,0 +1,214 @@
+)
+
+var (
+	log = golog.LoggerFor("authclient")
+)
+
+type authClient struct {
+	webclient webclient.RESTClient
+}
+
+type AuthClient interface {
+	//Sign up methods
+	SignUp(ctx context.Context, signupData *protos.SignupRequest) (bool, error)
+	SignupEmailResendCode(ctx context.Context, data *protos.SignupEmailResendRequest) (bool, error)
+	SignupEmailConfirmation(ctx context.Context, data *protos.ConfirmSignupRequest) (bool, error)
+
+	//Login methods
+	GetSalt(ctx context.Context, email string) (*protos.GetSaltResponse, error)
+	LoginPrepare(ctx context.Context, loginData *protos.PrepareRequest) (*protos.PrepareResponse, error)
+	Login(ctx context.Context, loginData *protos.LoginRequest) (*protos.LoginResponse, error)
+	// Recovery methods
+	StartRecoveryByEmail(ctx context.Context, loginData *protos.StartRecoveryByEmailRequest) (bool, error)
+	CompleteRecoveryByEmail(ctx context.Context, loginData *protos.CompleteRecoveryByEmailRequest) (bool, error)
+	ValidateEmailRecoveryCode(ctx context.Context, loginData *protos.ValidateRecoveryCodeRequest) (*protos.ValidateRecoveryCodeResponse, error)
+	// Change email methods
+	ChangeEmail(ctx context.Context, loginData *protos.ChangeEmailRequest) (bool, error)
+	// Complete change email methods
+	CompleteChangeEmail(ctx context.Context, loginData *protos.CompleteChangeEmailRequest) (bool, error)
+	DeleteAccount(ctc context.Context, loginData *protos.DeleteUserRequest) (bool, error)
+
+	//Logout
+	SignOut(ctx context.Context, logoutData *protos.LogoutRequest) (bool, error)
+}
+
+// NewClient creates a new instance of AuthClient
+func NewClient(baseURL string, opts *webclient.Opts) AuthClient {
+	httpClient := opts.HttpClient
+	if httpClient == nil {
+		httpClient = &http.Client{}
+	}
+	webclient := webclient.NewRESTClient(defaultwebclient.SendToURL(httpClient, baseURL, func(client *resty.Client, req *resty.Request) error {
+		req.SetHeader(common.ContentType, "application/x-protobuf")
+		if req.RawRequest.URL != nil && strings.HasPrefix(req.RawRequest.URL.Path, "/users/salt") {
+			// for the /users/salt endpoint, we do not need to set any headers so return right away
+			return nil
+		}
+		headers := map[string]string{}
+		uc := opts.UserConfig()
+		// Import all the internal headers
+		for k, v := range uc.GetInternalHeaders() {
+			headers[k] = v
+		}
+		req.SetHeaders(headers)
+		return nil
+	}, nil))
+	return &authClient{webclient}
+}
+
+// Auth APIS
+// GetSalt is used to get the salt for a given email address
+func (c *authClient) GetSalt(ctx context.Context, email string) (*protos.GetSaltResponse, error) {
+	var resp protos.GetSaltResponse
+	err := c.webclient.GetPROTOC(ctx, "/users/salt", map[string]interface{}{
+		"email": email,
+	}, &resp)
+	if err != nil {
+		return nil, err
+	}
+	return &resp, nil
+}
+
+// Sign up API
+// SignUp is used to sign up a new user with the SignupRequest
+func (c *authClient) SignUp(ctx context.Context, signupData *protos.SignupRequest) (bool, error) {
```

Comment:
on signUp, We still need to pass all default headers as we were passing before.

Reply:
Ah ok, thanks!

Reply:
I see `/signup` is the only `/users` endpoint where we require some of the default Pro headers https://github.com/getlantern/lantern-cloud/blob/main/cmd/api/users/mount.go#L45